kadi/scripts/validate_states.py

```diff
@@ -56,6 +67,132 @@ def get_opt():
+    days
+        length of validation interval (days)
+    states
+        list of states to validate (default=all)
+    no_exclude
+        if True then do not exclude intervals (default=False)
+
+    Returns
+    -------
+    list
+        List of validators
+    """
+    stop = CxoTime(stop)
+
+    validators = []
+    for cls in Validate.subclasses:
+        if states and cls.state_name not in states:
+            continue
+        logger.info(f"Validating {cls.state_name}")
+        validator: Validate = cls(stop=stop, days=days, no_exclude=no_exclude)
+        validator.html = validator.get_html()
+        validators.append(validator)
+
+    return validators
+
+
+def get_violations(validators: list[Validate]) -> list[dict]:
+    """Return list of violations from list of validators.
+
+    Parameters
+    ----------
+    validators : list
+        List of validators
+
+    Returns
+    -------
+    list
+        List of violations dicts, keys are "name", "start", "stop"
+    """
+    violations = [
+        {
+            "name": validator.state_name,
+            "start": violation["start"],
+            "stop": violation["stop"],
+        }
+        for validator in validators
+        for violation in validator.violations
+    ]
+    return violations
+
+
+def get_index_page_html(validators: list[Validate], violations: list[dict]) -> str:
+    """Make a simple HTML page with all the validation plots and information.
+
+    Parameters
+    ----------
+    validators : list[Validate]
+        List of processed validators
+    violations : list[dict]
+        List of violations dicts, keys are "name", "start", "stop"
+
+    Returns
+    -------
+    str
+        HTML string
+    """
+    context = {
+        "validators": validators,
+        "violations": violations,
+    }
+    index_template_file = (
+        Path(kadi.commands.__file__).parent / "templates" / "index_validate.html"
+    )
+    index_template = index_template_file.read_text()
+    template = jinja2.Template(index_template)
+    html = template.render(context)
+
+    return html
+
+
+def get_violations_text(violations: list[dict]) -> str:
+    """Get text for email or logger alert of validation violations.
+
+    Parameters
+    ----------
+    violations : list[dict]
+        List of violations dicts (see get_violations())
+
+    Returns
+    -------
+    str
+        Text for alert
+    """
+    lines = ["kadi validate_states processing found state violation(s):", ""]
+    lines.extend(Table(violations).pformat_all())
+    lines.extend(["", "See:https://cxc.harvard.edu/mta/ASPECT/validate_states/", ""])
+    text = "\n".join(lines)
+    return text
+
+
+def send_alert_email(text: str, opt: argparse.Namespace) -> None:
+    """Send an email alert for validation violations.
+
+    Parameters
+    ----------
+    text : str
+        Text for email
+    opt : argparse.Namespace
+        Command-line options
+    """
+    from acdc.common import send_mail
```

Comment:
acdc is presently a non-fsds package.  One of the reasons we let it go there was the idea that this acdc.common send mail import was *also* only used by other non-fsds packages.  So I see this new import in kadi validation as suggesting one of several paths:
1) Follow through and copy/move this send_mail into something like ska_helpers
2) Put acdc back into ska3-flight
3) Put this kadi validation and trending into a separate non-fsds package instead of kadi proper
4) Leave it like this and leave acdc in non-fsds packages but realize the validation reporting then depends on a package outside of flight (practically, the validation is likely going to be run from HEAD ska3/flight which will have acdc installed).

Reply:
Good point. Let's go with 4 right now. Tests should still pass without acdc installed. I have been bothered by having this utility living in acdc so this will get fixed in a reasonable time scale.